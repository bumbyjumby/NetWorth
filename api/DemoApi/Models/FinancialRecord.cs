using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace DemoApi.Models
{
    public class FinancialRecord
    {
        public int Id { get; set; }
        public double Value { get; set; }
        public string Currency { get; set; }
        public string Name { get; set; }
        public double Rate { get; set; }
        public double MonthlyPayment { get; set; }
        public RecordType RecordType { get; set; }
    }
}
