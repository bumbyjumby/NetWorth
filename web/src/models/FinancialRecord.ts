export enum RecordType{
    ShortTermAsset,
    LongTermAsset,
    ShortTermLiability,
    LongTermLiability
}

export default class FinancialRecord{
    id: number | undefined;
    value: number | undefined;
    // currency: string | undefined;
    name: string | undefined;
    rate: number | undefined;
    monthlyPayment: number | undefined;
    recordType: RecordType | undefined;
}