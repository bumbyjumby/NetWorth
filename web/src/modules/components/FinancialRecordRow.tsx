import React, { Component } from "react";
import { Grid, TextField, Input, Typography, InputAdornment } from "@material-ui/core";
import FinancialRecord from "../../models/FinancialRecord";
export interface IFinancialRecordRowProps {
    currencySymbol: string;
    financialRecord: FinancialRecord,
    onChange: (financialRecord: FinancialRecord) => void,
    showPayment: boolean
}
export default class FinancialRecordRow extends Component<IFinancialRecordRowProps>{

    render() {
        const { financialRecord, onChange, currencySymbol } = this.props;
        return (
            <Grid item xs={12}>
                <Grid container spacing={16}>
                    <Grid item xs={8}>
                        <Typography variant="subtitle1">
                            {financialRecord.name}
                        </Typography>
                    </Grid>
                    <Grid item xs={2}>
                        <Input
                            value={financialRecord.rate}
                            onChange={(event) => { financialRecord.rate = parseFloat(event.target.value); onChange(financialRecord) }}
                        /></Grid>
                    <Grid item xs={2}>
                        <TextField
                            value={financialRecord.value as number}
                            onChange={(event) => { financialRecord.value = parseFloat(event.target.value); onChange(financialRecord) }}
                            InputProps={{
                                startAdornment: <InputAdornment position="start">{currencySymbol}</InputAdornment>,
                              }}
                        />
                    </Grid>
                </Grid> 
            </Grid>
        )
    }
}