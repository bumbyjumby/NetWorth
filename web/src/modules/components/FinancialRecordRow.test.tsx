import React from 'react';
import ReactDOM from 'react-dom';

import FinancialRecordRow from './FinancialRecordRow';
import { shallow, render, mount } from 'enzyme';
import Select from 'react-select/lib/Select';
import renderer from 'react-test-renderer';
import FinancialRecord, { RecordType } from '../../models/FinancialRecord';

const financialRecord: FinancialRecord = {
    id:0,
    monthlyPayment:100,
    name:'stuff',
    rate:0.5,
    recordType: RecordType.ShortTermAsset,
    value:1000
}
function onChange(){}

it('renders correctly and creates a snapshot', () => {
    const tree = renderer.create(<FinancialRecordRow showPayment={true}
        currency={'CDN'}
        financialRecord={financialRecord}
        onChange={onChange} />).toJSON();
    expect(tree).toMatchSnapshot();
    
});