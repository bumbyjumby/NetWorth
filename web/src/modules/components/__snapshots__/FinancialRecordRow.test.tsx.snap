// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly and creates a snapshot 1`] = `
<div
  className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
>
  <div
    className="MuiGrid-container-1 MuiGrid-spacing-xs-16-24"
  >
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-8-37"
    >
      <h6
        className="MuiTypography-root-98 MuiTypography-subtitle1-116"
      >
        stuff
      </h6>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-2-31"
    >
      <div
        className="MuiInputBase-root-147 MuiInput-root-134 MuiInput-underline-138"
        onClick={[Function]}
      >
        <input
          className="MuiInputBase-input-157 MuiInput-input-142"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="text"
          value={0.5}
        />
      </div>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-2-31"
    >
      <div
        className="MuiInputBase-root-147 MuiInput-root-134 MuiInput-underline-138"
        onClick={[Function]}
      >
        <input
          className="MuiInputBase-input-157 MuiInput-input-142"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="text"
          value="1000.00"
        />
      </div>
    </div>
  </div>
</div>
`;
